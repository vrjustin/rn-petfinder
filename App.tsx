import React, {useEffect, useState} from 'react';
import {SafeAreaView, Text, View, FlatList} from 'react-native';
import getPetTypes from './services/apiService';
import PetType from './models/PetType';

function App(): React.JSX.Element {
  const [petTypes, setPetTypes] = useState<PetType[]>([]);

  useEffect(() => {
    const fetchPetTypesData = async () => {
      const typesData = await getPetTypes();
      setPetTypes(typesData?.types ?? []);
    };

    fetchPetTypesData();
  }, []);

  const renderItem = ({item}: {item: PetType}) => (
    <View>
      <Text>{item.name}</Text>
    </View>
  );

  return (
    <SafeAreaView>
      <View>
        <Text>Select an Animal Type to Get Started:</Text>
        <FlatList
          data={petTypes}
          renderItem={renderItem}
          keyExtractor={item => item.name}
        />
      </View>
    </SafeAreaView>
  );
}

export default App;
