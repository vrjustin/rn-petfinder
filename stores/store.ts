import {combineReducers, configureStore} from '@reduxjs/toolkit';
import petTypesReducer from '../reducers/petTypesReducer';
import petBreedsReducer from '../reducers/petBreedsReducer';
import animalsReducer from '../reducers/animalsReducer';
import searchParamsReducer from '../reducers/searchParamsReducer';

const createDebugger = require('redux-flipper').default;

const rootReducer = combineReducers({
  petTypes: petTypesReducer,
  petBreeds: petBreedsReducer,
  animals: animalsReducer,
  searchParameters: searchParamsReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware =>
    __DEV__
      ? getDefaultMiddleware({serializableCheck: false}).concat(
          createDebugger(),
        )
      : getDefaultMiddleware({serializableCheck: false}),
});

export default store;
