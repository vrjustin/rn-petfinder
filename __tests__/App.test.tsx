/**
 * @format
 */

import 'react-native';
import React from 'react';
import store from '../stores/store';
import App from '../App';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import {Provider} from 'react-redux';

jest.mock('../services/authenticationServices.ts', () => ({
  configureGoogleSignin: jest.fn(),
  configureOkta: jest.fn(),
  signOutOkta: jest.fn(),
  signOutGoogle: jest.fn(),
  signOutGuest: jest.fn(),
  handleSignInViaGoogle: jest.fn(),
  handleSignInViaGuest: jest.fn(),
  handleSignIn: jest.fn(),
  fetchToken: jest.fn(() => Promise.resolve(null)),
}));

jest.mock('../Views/PetTypes.tsx');

let renderedAppTree: any;

it('renders correctly', async () => {
  await act(async () => {
    renderedAppTree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
  });
  expect(renderedAppTree.toJSON()).toMatchSnapshot();
});
