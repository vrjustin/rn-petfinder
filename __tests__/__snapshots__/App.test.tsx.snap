// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LEGACY_RNCViewPager
  initialPage={0}
  layoutDirection="ltr"
  onMoveShouldSetResponderCapture={[Function]}
  onPageScroll={[Function]}
  onPageScrollStateChanged={[Function]}
  onPageSelected={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="Onboarding-PagerView"
  useLegacy={true}
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "flex": 1,
            "justifyContent": "flex-start",
          },
          {
            "backgroundColor": "#C5C5C5",
          },
        ]
      }
    >
      <Image
        source={
          {
            "default": "",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            {
              "resizeMode": "contain",
            },
          ]
        }
      />
      <RCTSafeAreaView
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "paddingTop": "20%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 24,
                "marginBottom": 20,
              }
            }
          >
            Welcome to PetFinder
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingTop": "20%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 24,
                "marginBottom": 20,
              }
            }
          >
            Swipe to Continue...
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "flex": 1,
            "justifyContent": "flex-start",
          },
          {
            "backgroundColor": "#B75D39",
          },
        ]
      }
    >
      <Image
        source={
          {
            "default": "",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            {
              "resizeMode": "contain",
            },
          ]
        }
      />
      <RCTSafeAreaView
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "paddingTop": "20%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 24,
                "marginBottom": 20,
              }
            }
          >
            Where all the Pets are Found!
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingTop": "20%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 24,
                "marginBottom": 20,
              }
            }
          >
            Adopt, don't Shop!
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "flex": 1,
            "justifyContent": "flex-start",
          },
          {
            "backgroundColor": "#28AEC7",
          },
        ]
      }
    >
      <Image
        source={
          {
            "default": "",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            {
              "resizeMode": "contain",
            },
          ]
        }
      />
      <RCTSafeAreaView
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "paddingTop": "20%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 24,
                "marginBottom": 20,
              }
            }
          >
            Ready to Find your Pet?
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-end",
              "paddingTop": "20%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 24,
                "marginBottom": 20,
              }
            }
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "rgba(255,255,255,0.7)",
                "borderRadius": 5,
                "opacity": 1,
                "padding": 10,
              }
            }
          >
            <Text>
              Continue to App
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</LEGACY_RNCViewPager>
`;
